v2/component/launcher.go

```diff
@@ -106,6 +106,16 @@ func (o *LauncherOptions) validate() error {
 	if empty(o.MLMDServerPort) {
 		return err("MLMDServerPort")
 	}
+	if strings.HasPrefix(o.PipelineName, "namespace/") {
+		s := strings.Split(o.PipelineName, "/")
+		if len(s) != 4 {
+			return fmt.Errorf("invalid PipelineName options for namespaced pipelines, need to follow 'namespace/${namespace}/pipeline/${pipelineName}': %s", o.PipelineName)
+		}
+		namespace := s[1]
+		if namespace != o.Namespace {
+			return fmt.Errorf("the namespace %s extracted from pipelineName is not equal to the namespace %s in launcher options", namespace, o.Namespace)
+		}
+	}
```

Comment:
For discussion: is it helpful adding the validation in launcher? I feel like launcher can just be unaware of the pipeline name format, if there's a problem. It'll be reported at task server.

Reply:
Info like "PipelineName"  and "Namespace" will get recorded in MLMD when cache is not enabled.  I think launcher should also be aware of the validation.